### PR TITLE
#13398 Add sorting options for media folders, which put photos or videos first.

### PR DESCRIPTION
**Pull request by mattw-mega**

### CODE REVIEW

File: src/megaclient.cpp [hunk 1]
```diff
@@ -13779,27 +13779,94 @@ node_vector MegaClient::getRecentNodes(unsigned maxcount, m_time_t since, bool i
 
 namespace action_bucket_compare
 {
-    std::mutex media_check;   // when we can use c++11, switch to a lambda that remembers the MegaClient* for compare().  In the meantime, force just one MegaClient at a time instead.
-    MegaClient* mc;
-
     const static string webclient_is_image_def = ".jpg.jpeg.gif.bmp.png.";
```
Comment (cb-mega) [inline src/megaclient.cpp]: Add a comment above this along the lines of: "Don't edit these lists. If you have to then keep them in sync with the webclient!"

File: include/megaapi_impl.h [hunk 2]
```diff
@@ -1834,7 +1834,7 @@ class SearchTreeProcessor : public TreeProcessor
 class OutShareProcessor : public TreeProcessor
 {
     public:
-        OutShareProcessor();
+        OutShareProcessor(MegaClient*);
```
[the remark this answers is not in the document]
Comment (mattw-mega) [reply include/megaapi_impl.h]: done!

File: include/mega/megaclient.h [hunk 3]
```diff
@@ -1153,6 +1153,11 @@ class MEGA_API MegaClient
     // get a vector of recent actions in the account
     recentactions_vector getRecentActions(unsigned maxcount, m_time_t since);
 
+    // determine if the file is a video, photo, or media (video or photo).  If the extension (with trailing .) is not precalculated, pass null
+    bool nodeIsPhoto(const Node*, char ext[12] = nullptr);
+    bool nodeIsVideo(const Node*, char ext[12] = nullptr);
+    bool nodeIsMedia(const Node*, bool* isphoto, bool* isvideo, char ext[12] = nullptr);
+
```
[the remark this answers is not in the document]
Comment (mattw-mega) [reply include/mega/megaclient.h]: removed the unused ones, and const does actually work

File: src/megaclient.cpp (hunk 1 above)
Comment (mattw-mega) [reply src/megaclient.cpp]: done!

File: src/megaapi_impl.cpp [hunk 4]
```diff
@@ -16400,25 +16394,35 @@ int naturalsorting_compare (const char *i, const char *j)
     return 0;
 }
 
-std::function<bool (Node*, Node*)>MegaApiImpl::getComparatorFunction(int order)
+std::function<bool (Node*, Node*)> MegaApiImpl::getComparatorFunction(int order, MegaClient* mc)
 {
-    std::function<bool (Node*, Node*)> comp;
     switch(order)
```
[the remark this answers is not in the document]
Comment (mattw-mega) [reply src/megaapi_impl.cpp]: done!

**Pull request merged**